src/main/scala/org/apache/spark/clueso/metrics/SearchMetricsSource.scala

```diff
@@ -0,0 +1,147 @@
+package org.apache.spark.clueso.metrics
```

Comment:
I'm having a hard time following the code in this file so whatever comments you could add would be appreciated.